lib/vnet/user_process_darwin.go

```diff
@@ -18,121 +18,94 @@ package vnet
 
 import (
 	"context"
-	"errors"
-	"time"
+	"net"
+	"os"
 
 	"github.com/gravitational/trace"
 	"github.com/jonboulle/clockwork"
-	"golang.zx2c4.com/wireguard/tun"
+	"google.golang.org/grpc"
+	grpccredentials "google.golang.org/grpc/credentials"
 
+	"github.com/gravitational/teleport/api/utils/grpc/interceptors"
+	vnetv1 "github.com/gravitational/teleport/gen/proto/go/teleport/lib/vnet/v1"
 	"github.com/gravitational/teleport/lib/vnet/daemon"
 )
 
-// runPlatformUserProcess creates a network stack for VNet and runs it in the
-// background. To do this, it also needs to launch an admin process in the
-// background. It returns a [ProcessManager] which controls the lifecycle of
-// both background tasks.
-func runPlatformUserProcess(ctx context.Context, cfg *UserProcessConfig) (pm *ProcessManager, nsi NetworkStackInfo, err error) {
-	// Make sure to close the process manager if returning a non-nil error.
-	defer func() {
-		if pm != nil && err != nil {
-			pm.Close()
-		}
-	}()
+// runPlatformUserProcess launches a daemon in the background that will handle
+// all networking and OS configuration. The user process exposes a gRPC
+// interface that the daemon uses to query application names and get user
+// certificates for apps. It returns a [ProcessManager] which controls the
+// lifecycle of both the user and daemon processes.
+func runPlatformUserProcess(ctx context.Context, cfg *UserProcessConfig) (*ProcessManager, *vnetv1.NetworkStackInfo, error) {
+	ipcCreds, err := newIPCCredentials()
+	if err != nil {
+		return nil, nil, trace.Wrap(err, "creating credentials for IPC")
+	}
+	serverTLSConfig, err := ipcCreds.server.serverTLSConfig()
+	if err != nil {
+		return nil, nil, trace.Wrap(err, "generating gRPC server TLS config")
+	}
 
-	ipv6Prefix, err := newIPv6Prefix()
+	credDir, err := os.MkdirTemp("", "vnet_service_certs")
 	if err != nil {
-		return nil, nsi, trace.Wrap(err)
+		return nil, nil, trace.Wrap(err, "creating temp dir for service certs")
+	}
+	// Write credentials with 0200 so that only root can read them and no user
+	// processes should be able to connect to the service.
+	if err := ipcCreds.client.write(credDir, 0200); err != nil {
+		return nil, nil, trace.Wrap(err, "writing service IPC credentials")
 	}
-	dnsIPv6 := ipv6WithSuffix(ipv6Prefix, []byte{2})
 
-	// Create the socket that's used to receive the TUN device from the admin process.
-	socket, socketPath, err := createSocket()
+	listener, err := net.Listen("tcp", "localhost:0")
```

Comment:
Why do we not use a unix socket here? I think you might've mentioned it already but if so I managed to forget it. 🙊

Reply:
we actually could on mac (and linux later) i just did the same here as windows